### PR TITLE
Added PCH and Refactor

### PR DESCRIPTION
**Pull request by NOOBDY**
No functional changes, only setup precompiled headers to improve compile times and refactored some code to provide a better base to build on.

### CODE REVIEW

File: src/main.cpp
Comment (ntut-rick) [inline src/main.cpp:1]: sus

**Review by ntut-rick — APPROVED** [review]
LGTM

File: src/main.cpp
Comment (NOOBDY) [reply src/main.cpp:1]: no need for std::cout because we have logger system now

**Review by Dragonfly911117 — APPROVED** [review]
LGTM

**Pull request merged**